.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,241 @@
+name: build-and-publish
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+env:
+  snap7-archive-url: https://sourceforge.net/projects/snap7/files/1.4.2/snap7-full-1.4.2.7z/download
+jobs:
+  linux-build:
+    name: Build binary wheels for linux
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - arch: x86_64
+            makefile: x86_64_linux.mk
+          - arch: i686
+            makefile: i386_linux.mk
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Debian packages
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install -y python3-pip python3-setuptools p7zip
+
+      - name: Cache snap7-archive
+        id: snap7-archive
+        uses: actions/cache@v2
+        with:
+          path: snap7-full-1.4.2.7z
+          key: ${{ env.snap7-archive-url }}
+
+      - name: Get snap7
+        if: steps.snap7-archive.outputs.cache-hit != 'true'
+        run: |
+          wget -O snap7-full-1.4.2.7z --content-disposition -c ${{ env.snap7-archive-url }}
+
+      - name: Extract archive
+        run: 7z x snap7-full-1.4.2.7z
+
+      - name: copy files
+        run: |
+          mkdir src
+          cp snap7-full-1.4.2/release/Wrappers/c-cpp/snap7.h src
+
+      - name: Install cibuildwheel and wheel
+        run: python3 -m pip install cibuildwheel==1.10.0 wheel --upgrade
+
+      - name: Build wheels linux
+        run: python3 -m cibuildwheel --output-dir wheelhouse
+        env:
+          CIBW_BUILD: cp3*-manylinux_${{ matrix.arch }}
+          CIBW_SKIP: cp35-* pp*
+          CIBW_BEFORE_ALL_LINUX: cd snap7-full-1.4.2/build/unix/;make -f ${{ matrix.makefile }} install
+
+      - name: move shared library to ./lib
+        run: |
+          cd wheelhouse
+          for wheelname in $(ls *.whl)
+          do
+          dirname=$wheelname-dir
+          packagename=$(python3 -m wheel unpack $wheelname -d $dirname | sed -e 's/.*\/\(.*\)...OK/\1/')
+          mkdir -p $dirname/$packagename/snap7/lib
+          mv $dirname/$packagename/python_snap7.libs/libsnap7-*.so $dirname/$packagename/snap7/lib/libsnap7.so
+          rm -rf $dirname/$packagename/python_snap7.libs
+          rm $dirname/$packagename/snap7/__dummy__*.so
+          python3 -m wheel pack $dirname/$packagename -d ./
+          rm -rf $dirname
+          done
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: wheelhouse/*.whl
+
+  windows-build:
+    name: Build binary wheels for windows
+    runs-on: windows-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - arch: win_amd64
+            dll_dir: Win64
+          - arch: win32
+            dll_dir: Win32
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install choco packages
+        run: choco install 7zip wget python
```

Comment:
so you only build this for 3.9? Ideally, we do it for 3.6, 3.7 and 3.8 also...

I've managed to get multiple pythons installed with choco on windows:

https://github.com/MonetDBSolutions/MonetDBe-Python/blob/master/.github/workflows/windows.yml#L24

but it is not the most stable solution.

Reply:
I build one package for all python versions. It looks like python_snap7-1.0**-py3**-none-win_amd64.whl. Pip considers this package to be suitable for any 3.* version of python in Windows x64.

Reply:
you are right! I didn't realise, but since we don't include anything compiled to the python library, we don't need to build for every python!  And i guess it is the same for all platforms actually. awesome.